### PR TITLE
chore: update contributing.md and setup

### PR DESCRIPTION
**Pull request by hehex9**
~~, I noticed that `pnpm i` changed the pnpm-lock.yaml but no error was displayed.~~

in this pr:
- ~~updates pnpm-lockfile~~ (magically resolved by recent commits, no idea why)
- It seems that `-r` is unnecessary inside pnpm workspace? (no sure)
- `pnpm run setup` includes `pnpm i` already I think
- found that the `platforms` array from `@prisma/get-form` can be used in sdk package.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -26,7 +26,6 @@ To setup and build the packages, follow these steps:
 git clone https://github.com/prisma/prisma.git
 npm i -g pnpm@6 --unsafe-perm
 cd prisma/src
-pnpm i
```
Comment (janpio) [inline CONTRIBUTING.md:29]: Why remove this?

File: src/scripts/setup.ts [hunk 2]
```diff
@@ -62,7 +62,7 @@ has to point to the dev version you want to promote, for example 2.1.0-dev.123`)
 
     await run(
       '.',
-      `pnpm i --no-prefer-frozen-lockfile -r --reporter=silent`,
```
Comment (janpio) [inline src/scripts/setup.ts:65]: Why this change?

File: src/scripts/setup.ts [hunk 3]
```diff
@@ -113,7 +113,7 @@ has to point to the dev version you want to promote, for example 2.1.0-dev.123`)
   // final install on top level
   await pRetry(
     async () => {
-      await run('.', 'pnpm i --no-prefer-frozen-lockfile -r')
```
Comment (janpio) [inline src/scripts/setup.ts:116]: Why this change?

**Review by millsp — APPROVED** [review]

**Comment by millsp** [comment]
I just had to revert `pnpm i` as it is needed. Thanks for your contribution @hehex9!

File: src/packages/sdk/src/getGenerators.ts [hunk 4]
```diff
@@ -543,25 +543,7 @@ export function skipIndex<T = any>(arr: T[], index: number): T[] {
...
-  'linux-musl',
-  'linux-nixos',
-  'windows',
-  'freebsd11',
-  'freebsd12',
-  'openbsd',
-  'netbsd',
-  'arm',
-]
+export const knownBinaryTargets: Platform[] = [...platforms, 'native']
```
Comment (Jolg42) [inline src/packages/sdk/src/getGenerators.ts:546]: 😍

Comment (Jolg42) [inline src/packages/sdk/src/getGenerators.ts:546]: This is neat!

Note platforms is defined here https://github.com/prisma/engines-wrapper/blob/master/packages/get-platform/src/platforms.ts
